Qa python homework 2

kontsovart opened the pull request:
# Домашнее задание №2

Прошу @snicks92 или @z1ll1 проверить его.

Что было сделано:
Проведено тестирование портала https://target.my.com/
Тесты разделены по маркам UI и API.
Запуск тестов по дефолту локально в стоковом браузере через webdriver_manager.
Если указан параметр --selenoid=host:port, тесты идут на selenoid (пример команды pytest -s -v hw2 -m API --selenoid=0.0.0.0:4444).
Тесты являются независимыми и запускаются параллельно при указании параметра -n.

По UI:
1) Позитивный и негативный тест на авторизацию
2) Фикстура, которая авторизовывается и возвращает объект главной страницы
3) Тест на создание рекламной комании (трафик, баннер)
4) Тест на создание сегмента в аудиториях и проверка создания сегмента
5) Тест на удаление сегмента после создания
6) Реализовано на модели PageObject

По API:
1) Клиент, у которого есть метод авторизации
2) Тест для метода авторизации
3) Тест на создание сегмента в аудиториях и проверка создания сегмента
4) Тест на удаление сегмента после создания

kontsovart committed:
initial commit
Done:
selenoid params
multiprocessing params
UI/API marks

UI(PageObject pattern): positive/negative auth test, auth fixture test, create new campaign test, create segment test, delete segment test
API: Client with auth method, Create segment method

kontsovart committed:
Done:
API:Create audience test, delete audience test
UI:New locator

SK1995 commented on hw2/pytest.ini:
```diff
@@ -0,0 +1,4 @@
+[pytest]
+markers =
+    UI: mark a test as a UI test
+    API: mark a test as a API test
```
Нужен символ переноса в конце строки.  
Почему он нужен можно посмотреть  [тут](https://stackoverflow.com/questions/5813311/no-newline-at-end-of-file).

kontsovart replied:
Добавлен символ переноса в конце строки.

SK1995 commented on hw2/tests/test_api.py:
```diff
@@ -0,0 +1,61 @@
+
+        return MyTargetClient(user, password)
+
+    # @pytest.mark.skip(reason='TEMP')
```
Выглядит как отладочный комментарий, такие комментарии лучше убирать из финальной версии  PR.

kontsovart replied:
Комментарии были удалены.

SK1995 commented on hw2/tests/test_api.py:
```diff
@@ -0,0 +1,61 @@
+
+class TestMyTarget:
+
+    @pytest.fixture(scope='function')
```
Здесь лучше использовать скоуп сессии, чтобы не создавать нового апи клиента на каждый тест, который его использует.

kontsovart replied:
Заменено на скоуп сессии.

SK1995 commented on hw2/tests/test_api.py:
```diff
@@ -0,0 +1,61 @@
+    def test_auth(self, api_client):
+        api_client.auth_post()
+        response = api_client.get_cabinet_page()
+        api_client.get_z_cookie_page()
```
А этот запрос тут точно нужен?

kontsovart replied:
Данный запрос не нужен и удален.

SK1995 commented on hw2/tests/test_api.py:
```diff
@@ -0,0 +1,61 @@
+    # @pytest.mark.skip(reason='TEMP')
+    @pytest.mark.API
+    def test_new_segment(self, random_string_ascii, api_client):
+        api_client.auth_post()
```
Можно перенести авторизацию и проставление кук в какой-нибудь специальный метода типа `login` и вызывать его при создании объекта апи клиента.  
Так код тестов сразу сильно уменьшится и дублирования в них станет меньше.

kontsovart replied:
Добавлен новый метод авторизации и проставления кук с названием login.

SK1995 commented on hw2/ui/pages/login.py:
```diff
@@ -0,0 +1,7 @@
+
+
+class LoginPage(BasePage):
+    locators = LoginPageLocators()
```
Тоже надо новую строчку добавить

kontsovart replied:
Добавлен символ переноса в конце строки.

SK1995 reviewed (commented):
Все ок, есть мелкие комментарии

kontsovart committed:
Done:
Fixed some details: add api client scope session, delete comments, delete get z token
UI:New locator

kontsovart commented:
Замечания исправлены, работа готова к проверке.

SK1995 reviewed (approved):
Хорошая работа! =)  
Поставил максимум (18 баллов)
